### ISSUE TITLE
azapi_update_resource - unable to update VMSS that requires PATCH operation

### ISSUE THREAD

**Issue by mrowken** [Azure/terraform-provider-azapi#125]
I'm trying to assign User Assigned Managed Identity to AKS VMSS using Terraform.
It is working fine with AZ CLI method:
```bash
az vmss identity assign -g $vmssRG -n $vmssName --identities $id
```
Azure CLI uses HTTP PATCH and request body contains just single property "identity" instead of whole object.

Example of TF code:
```terraform
data "azurerm_resources" "vmss" {
  type = "Microsoft.Compute/virtualMachineScaleSets"
  resource_group_name = <node_resource_group>
  required_tags = {
    aks-managed-poolName = "nodepoolname"
  }
}

resource "azapi_update_resource" "vmss_identity" {
  type        = "Microsoft.Compute/virtualMachineScaleSets@2021-11-01"
  resource_id = data.azurerm_resources.vmss.resources.0.id

  body = jsonencode({    
    identity = {
      type = "UserAssigned"
      userAssignedIdentities = {
        "/subscriptions/<subscr_id>/resourceGroups/<rg_name>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/<identity_name>" = {}        
      }
    }
  })
}
```
Using this provider I got error during apply:
`however the current tenant '***' is not authorized to access linked subscription '109a5e88-712a-48ae-9078-9ca8b3c81345`
The root cause is well described in this issue:
https://github.com/Azure/AKS/issues/1819

I found in implementation there is fixed method PUT used:
https://github.com/Azure/terraform-provider-azapi/blob/5b23676cb92cf92b882108c89ad8ab6740b28a06/internal/clients/resource_client.go#L84

Could you provide a way to do this like some parameter to decide which HTTP method to use or just use PATCH for whole group of resources "Microsoft.Compute/" ?

**Comment by ms-henglu** [comment]
Hi @mrowken ,

Thank you for taking time to report this issue!

If the vm scale set is managed by azurerm provider, It's not recommended to use `azapi_update_resource` to manage some features which are already supported in azurerm, because it will cause diff between config and state.

Unfortunately, I couldn't reproduce this, it works for me. I created vmss under sub1. Under sub2, I created an user assigned identity, and successfully attached it to the vmss. Maybe check whether your credential has permission to access both subscriptions? Or does the subscriptions belong to 2 different tenants? If so, `auxiliary_tenant_ids` is not supported currently:
https://github.com/Azure/terraform-provider-azapi/blob/main/internal/provider/provider.go#L63

```
// config for sub2
resource "azurerm_resource_group" "test" {
  name     = "acctestRG-henglu"
  location = "west europe"
}

resource "azurerm_user_assigned_identity" "test" {
  name                = "henglu-api"
  resource_group_name = azurerm_resource_group.test.name
  location            = azurerm_resource_group.test.location
}

resource "azapi_update_resource" "test" {
  type        = "Microsoft.Compute/virtualMachineScaleSets@2021-11-01"
  resource_id = "/subscriptions/{sub1}/resourceGroups/acctestRG-henglu/providers/Microsoft.Compute/virtualMachineScaleSets/acctestvmss-henglu"
  body = jsonencode({
    identity = {
      type = "UserAssigned"
      userAssignedIdentities = {
        (azurerm_user_assigned_identity.test.id) = {}
      }
    }
  })
}

```

**Comment by mrowken** [comment]
Hi @ms-henglu,
Please notice important difference - my issue is related to VMSS created by AKS service (representing Nodepool).
That's why I didn't create VMSS myself and I don't manage it from Terraform. It is kind of filling gap in AKS + azurerm provider to use [AAD POD Identity](https://github.com/Azure/aad-pod-identity) in [managed mode](https://azure.github.io/aad-pod-identity/docs/configure/pod_identity_in_managed_mode/) as there is currently no option to add UAMI to VMSS via AKS (it would be the best option), but only on VMSS level.
I want to just assign identity to existing AKS VMSS after cluster creation/modification. Importing this resource into Terraform and managing using azurerm provider doesn't make sense, because it will be managed by AKS - conflicting changes expected.

Such VMSS has below specification (relevant part):
```json
"properties": {
	"virtualMachineProfile": {
		"storageProfile": {
			"imageReference": {
				"id": "/subscriptions/109a5e88-712a-48ae-9078-9ca8b3c81345/resourceGroups/AKS-Ubuntu/providers/Microsoft.Compute/galleries/AKSUbuntu/images/1804gen2containerd/versions/2022.05.10"
			},
		}
	}
}
```
VM image reference comes from some MS subscription that none of customers will have access to - that's why as long as you use PUT operation updating whole resource, Azure attempts to check your permission to this "Linked subscription". If this part would be omitted and PATCH operation used, it will work.

**Comment by ms-henglu** [comment]
Hi @mrowken - Thanks for the clarification, I'm working on a new resource `azapi_operation` which can provide more flexibility, it can help with this case. I'll keep you updated.

**Comment by hoivikaj** [comment]
@ms-henglu 

Just wanted to highlight another scenario that requires PATCH method. When attempting to assign users to Microsoft.DesktopVirtualization/hostPools/sessionHosts, PATCH is required. When trying to do this today, 405 Method not allowed for PUT.

Like the scenario above, all sessionHost operations are currently not available in the latest azurerm provider, and are even missing from azure cli - only present in azure powershell.

**Comment by ms-henglu** [comment]
Hi @mikemadeja ,

Thank you for the example! Yes, there're resources requiring PATCH method to update a resource.

**Comment by ms-henglu** [comment]
Hi @mrowken, @mikemadeja  ,

The [azapi_resource_action](https://registry.terraform.io/providers/Azure/azapi/latest/docs/resources/azapi_resource_action) has been released in azapi 0.5.0, would you please give it a try? And there's an example:
https://github.com/Azure/terraform-provider-azapi/blob/main/examples/Microsoft.Compute/add_UAI_to_VirtualMachineScaleSets_created_by_AKS/main.tf
